IPython/core/magics/execution.py

```diff
@@ -1036,43 +1060,29 @@ def timeit(self, line='', cell=None):
         # This is used to check if there is a huge difference between the
         # best and worst timings.
         # Issue: https://github.com/ipython/ipython/issues/6471
-        worst_tuning = 0
         if number == 0:
             # determine number so that 0.2 <= total time < 2.0
-            number = 1
-            for _ in range(1, 10):
+            for index in range(0, 10):
+                number = 10 ** index
                 time_number = timer.timeit(number)
-                worst_tuning = max(worst_tuning, time_number / number)
                 if time_number >= 0.2:
                     break
-                number *= 10
-        all_runs = timer.repeat(repeat, number)
-        best = min(all_runs) / number
 
-        worst = max(all_runs) / number
-        if worst_tuning:
-            worst = max(worst, worst_tuning)
+        all_runs = timer.repeat(repeat, number)
+        timeit_result = TimeitResult(number, repeat, all_runs, tc, precision)
 
         if not quiet :
             # Check best timing is greater than zero to avoid a
             # ZeroDivisionError.
             # In cases where the slowest timing is lesser than a micosecond
             # we assume that it does not really matter if the fastest
             # timing is 4 times faster than the slowest timing or not.
-            if worst > 4 * best and best > 0 and worst > 1e-6:
-                print("The slowest run took %0.2f times longer than the "
```

Comment:
I think I would leave this warning message. It is relatively recent, and was requested because users were regularly getting cached results.